### DISCUSSION TITLE
How to listen for SEI Data With Video.js

### DISCUSSION THREAD

**Discussion by lastpeony** [videojs/video.js#8731] [Q&A]
I am looking for a way to listen for SEI data with video.js
For ID3 its fairly simple i do it like this:

```
this.videojsPlayer.textTracks().on('addtrack', e => {
            const metadataTrack = Array.from(this.videojsPlayer.textTracks()).find(t => t.label === 'Timed Metadata');         
            if (metadataTrack) {
                metadataTrack.addEventListener('cuechange', () => {
                    console.log(metadataTrack.activeCues[0]?.text, metadataTrack.activeCues);
                });
            }
```

On HLS.js i can capture the SEI data easly with this:

```
 hls.on(Hls.Events.FRAG_PARSING_USERDATA, function (event, data) {
        console.log('event =====> ', event, 'data =====> ', data.samples[0].userData)
    });
```

How can i do the same in videojs? addtrack is not triggered when i add sei data to the stream. played around with it but couldnt find the data anywhere

**Comment by lastpeony** [comment]
Is there anyone who knows to do this?
Sorry for bothering @mister-ben  maybe you have an idea?

**Reply by mister-ben** [reply]
Do you have an example stream with data to test something?

**Reply by lastpeony** [reply]
Hello i am adding zip which has sei embedded hls stream.
[test-stream-with-sei.zip](https://github.com/videojs/video.js/files/15373405/test-stream-with-sei.zip)
thank you for your help

**Reply by lastpeony** [reply]
Hello @mister-ben did you have chance to test it? If any problem with stream let me know.
Thanks

**Reply by lastpeony** [reply]
Hello @mister-ben 
Did you have chance to check the stream files?
Thank you for your interest.

**Comment by lastpeony** [comment]
Does anyone have any ideas? It's very easy to do this with hls.js, but it's strange that there's literally no information about it online for video.js

**Comment by lastpeony** [comment]
I am still looking for answer.
